packages/caravan-health/src/wallet.ts

```diff
@@ -0,0 +1,141 @@
+      utxoSetLength += addressUtxos.length;
+    }
+    let utxoMassFactor: number;
+    if (utxoSetLength >= 50) {
+      utxoMassFactor = 0;
+    } else if (utxoSetLength >= 25 && utxoSetLength <= 49) {
+      utxoMassFactor = 0.25;
+    } else if (utxoSetLength >= 15 && utxoSetLength <= 24) {
+      utxoMassFactor = 0.5;
+    } else if (utxoSetLength >= 5 && utxoSetLength <= 14) {
+      utxoMassFactor = 0.75;
+    } else {
+      utxoMassFactor = 1;
+    }
+    return utxoMassFactor;
+  }
+
+  /*
+    Utility function that helps to obtain the fee rate of the transaction
+  */
+  getFeeRateForTransaction(transaction: Transaction): number {
+    let fees: number = transaction.fee;
+    let weight: number = transaction.weight;
+    return fees / weight;
+  }
+
+  /*
+    Utility function that helps to obtain the percentile of the fees paid by user in tx block
+  */
+  getFeeRatePercentileScore(
+    timestamp: number,
+    feeRate: number,
+    feeRatePercentileHistory: FeeRatePercentile[],
+  ): number {
+    let percentile: number = this.getClosestPercentile(
+      timestamp,
+      feeRate,
+      feeRatePercentileHistory,
+    );
+    return 1 - percentile / 100;
+  }
+
+  /*
+    Utility function that helps to obtain the closest percentile of the fees paid by user in tx block
+  */
+  getClosestPercentile(
+    timestamp: number,
+    feeRate: number,
+    feeRatePercentileHistory: FeeRatePercentile[],
+  ): number {
+    // Find the closest entry by timestamp
+    let closestBlock: FeeRatePercentile | null = null;
+    let closestDifference: number = Infinity;
+
+    for (const block of feeRatePercentileHistory) {
+      const difference = Math.abs(block.timestamp - timestamp);
+      if (difference <= closestDifference) {
+        closestDifference = difference;
+        closestBlock = block;
+      }
+    }
+    if (!closestBlock) {
+      throw new Error("No fee rate data found");
+    }
+    // Find the closest fee rate percentile
+    switch (true) {
+      case feeRate <= closestBlock.avgFee_0:
+        return 0;
+      case feeRate <= closestBlock.avgFee_10:
+        return 10;
+      case feeRate <= closestBlock.avgFee_25:
+        return 25;
+      case feeRate <= closestBlock.avgFee_50:
+        return 50;
+      case feeRate <= closestBlock.avgFee_75:
+        return 75;
+      case feeRate <= closestBlock.avgFee_90:
+        return 90;
+      case feeRate <= closestBlock.avgFee_100:
+        return 100;
+      default:
+        throw new Error("Invalid fee rate");
+    }
+  }
+
+  constructAddressUsageMap(): AddressUsageMap {
+    let addressUsageMap: AddressUsageMap = {};
+    const transactions = this.transactions;
+    for (const tx of transactions) {
+      for (const output of tx.vout) {
+        let address = output.scriptPubkeyAddress;
+        if (addressUsageMap[address]) {
+          addressUsageMap[address] = true;
+        } else {
+          addressUsageMap[address] = false;
+        }
```

Comment:
does this actually work? if the address doesn't exist in the map then the value is falsey. But if it does exist then the value will be `false` in which case the conditional check would also fail. So you would never have this set to true. 

If I'm correct about this then please make sure the tests are correct since I would have expected several tests to fail. 

To fix this, you can use an actual `Map` which has methods like `has` which wouldn't return false if the value exists. I also think this property is a little confusingly named. Since if you find an output with this address it means it's been used. So really this is meant to track multiple use, aka reuse right? Alternatively you can use a num instead of a bool to track the count and then the reuse can check if the number is greater than 1.